packages/dynamite/dynamite/lib/src/models/discriminator.dart

```diff
@@ -1,30 +1,16 @@
-import 'package:collection/collection.dart';
-import 'package:json_annotation/json_annotation.dart';
-import 'package:meta/meta.dart';
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
 
 part 'discriminator.g.dart';
 
-@JsonSerializable()
-@immutable
-class Discriminator {
-  const Discriminator({
-    required this.propertyName,
-    this.mapping,
-  });
+abstract class Discriminator implements Built<Discriminator, DiscriminatorBuilder> {
+  factory Discriminator([final void Function(DiscriminatorBuilder) updates]) = _$Discriminator;
 
-  factory Discriminator.fromJson(final Map<String, dynamic> json) => _$DiscriminatorFromJson(json);
-  Map<String, dynamic> toJson() => _$DiscriminatorToJson(this);
+  Discriminator._();
 
-  final String propertyName;
+  static Serializer<Discriminator> get serializer => _$discriminatorSerializer;
 
-  final Map<String, String>? mapping;
+  String get propertyName;
 
-  @override
-  bool operator ==(final Object other) =>
-      other is Discriminator &&
-      propertyName == other.propertyName &&
-      const MapEquality().equals(mapping, other.mapping);
-
-  @override
-  int get hashCode => propertyName.hashCode + const MapEquality().hash(mapping);
+  Map<String, String>? get mapping;
```

Comment:
I think to gain real collection equality we will need to use built_collection (`BuiltMap`, `BuiltList`, `BuiltSet`) throughout the models.

Also guarantees immutability to we don't mutate our input in dynamite  by accident.